From e336ea36631e2be6d81e4a7d23dfef01abde2a31 Mon Sep 17 00:00:00 2001
From: W-Mark Kubacki <wmark@hurrikane.de>
Date: Fri, 31 Jan 2014 11:26:21 +0100
Subject: [PATCH] OpenSSL: deselect weak ciphers by default

Selects only HIGH ciphers, excluding RC4 and 3DES because:

Some OpenSSL versions still consider RC4 and 3DES a strong cipher.
---
 lib/vtls/openssl.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/vtls/openssl.h b/lib/vtls/openssl.h
index 07448b5..1701e68 100644
--- a/lib/vtls/openssl.h
+++ b/lib/vtls/openssl.h
@@ -95,7 +95,7 @@ void Curl_ossl_md5sum(unsigned char *tmp, /* input */
 #define curlssl_random(x,y,z) Curl_ossl_random(x,y,z)
 #define curlssl_md5sum(a,b,c,d) Curl_ossl_md5sum(a,b,c,d)
 
-#define DEFAULT_CIPHER_SELECTION "ALL!EXPORT!EXPORT40!EXPORT56!aNULL!LOW!RC4"
+#define DEFAULT_CIPHER_SELECTION "HIGH!aNULL!eNULL!RC4!3DES"
 
 #endif /* USE_SSLEAY */
 #endif /* HEADER_CURL_SSLUSE_H */
-- 
1.8.5.3

